client/actions/planning/api.ts

```diff
@@ -115,36 +117,133 @@ function query(
                         return Promise.resolve(result);
                     }
 
-                    return Promise.resolve(response._items);
+                    return Promise.resolve(response);
                 } else {
                     return Promise.reject('Failed to retrieve items');
                 }
             }, (error) => Promise.reject(error))
     );
 }
 
+const handleItemsForLastFetchedDay = (
+    items: Array<IPlanningItem>,
+    params: IPlanningSearchParams = {},
+    total: number,
+    dispatch: any,
+    getState: () => any,
+): Promise<Array<IPlanningItem>> => {
+    if (items.length < 1) {
+        return Promise.resolve([]);
+    }
+
+    const lastDayGroupItems = selectors.planning.lastDayGroup(getState()) ?? [];
+    const itemIdsInList = selectors.planning.planIdsInList(getState());
+
+    const hasFetchedTotalItems = total === (items ?? []).length + (lastDayGroupItems ?? []).length;
```

Comment:
imo this can only be true when you fetch all items on the first load (with optional loadMore call when you get single group), because you're ignoring the items fetched previously.
if you would also count items already rendered it should be ok and you wouldn't need that next one, which I wonder when it's actually true, probably when total is 0 or if it's calling the method after fetching 0 items (so you know you're at the end but still call it once more)